src/java/org/apache/cassandra/utils/bytecomparable/ByteComparable.java

```diff
@@ -39,6 +48,7 @@ enum Version
     {
         LEGACY,
         OSS41,  // CASSANDRA 4.1 encoding, used in trie-based indices
+        OSS50,  // CASSANDRA 5.0 encoding, used by the trie memtable
```

Comment:
In terms of SAI tries and `ByteComparable` encodings already persisted to disk, I think this new version will require us to declare a `ByteComparable` version used to encode a given trie. Is that correct? The current code has some hard coded references to `OSS41`.